### DISCUSSION TITLE
Erro de configuração

### DISCUSSION THREAD

**Discussion by anacarolinaluizsantos** [frontendbr/forum#2507] [Perguntas e Respostas]
Boa noite, tudo bem? De forma inesperada, ao refazer um curso de Git/GitHub e acessar as configurações, houve  a seguinte situação: 
![erro_git](https://github.com/user-attachments/assets/a19fac90-91dd-4b79-9a25-978611ceb2dd)
O que causou o erro e como resolver? Sou iniciante na área e não enbtendo quase nada de tecnologia.

**Comment by mateusmoura** [comment]
> Boa noite, tudo bem? De forma inesperada, ao refazer um curso de Git/GitHub e acessar as configurações, houve a seguinte situação: ![erro_git](https://private-user-images.githubusercontent.com/65691449/374355870-a19fac90-91dd-4b79-9a25-978611ceb2dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzNDUxNDMsIm5iZiI6MTcyODM0NDg0MywicGF0aCI6Ii82NTY5MTQ0OS8zNzQzNTU4NzAtYTE5ZmFjOTAtOTFkZC00Yjc5LTlhMjUtOTc4NjExY2ViMmRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA3VDIzNDcyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2NTNjMTg2ZjYzOTdlODM1ZmYzZWM2ZDg5Y2UzMjFiYjY1ZGIyNTBlNDc4MGY2MGIyYjI2YTRiYWE1NzE2YWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iRU9Ldj6LQO6UBaf29i5SL277uEHEbb6R6uAylyRWzY) O que causou o erro e como resolver? Sou iniciante na área e não enbtendo quase nada de tecnologia.

Olá @anacarolinaluizsantos 

Para executar o git config sempre é necessário passar uma ação! 

Exemplo: 

`git config --global user.name "Ana Carolina"`

Segue a doc sobre o git config: 

https://git-scm.com/docs/git-config

**Reply by anacarolinaluizsantos** [reply]
Obrigada, resolveu o problema.

**Comment by henriquetourinho** [comment]
## Fica tranquila, erro de configuração é normal no começo\!

Boa noite\! Pelo que parece na sua captura de tela, você está tentando rodar um comando do Git, mas **faltou informar o que você quer configurar**.

O erro **`error: no action specified`** significa exatamente isso: o Git (ou o sistema) não sabe qual é o próximo passo depois de `git config`.

Você mencionou que está refazendo um curso, e o comando `git config` é geralmente usado para configurar suas informações de usuário globalmente.

### 📝 O Comando Correto

Para o Git funcionar direito e registrar quem fez as alterações no código, você precisa informar seu nome e e-mail.

Tente rodar esses dois comandos, substituindo `SEU NOME` e `seu.email@exemplo.com` pelos seus dados:

#### 1\. Configurar o Nome do Usuário:

```bash
git config --global user.name "SEU NOME"
```

#### 2\. Configurar o E-mail do Usuário:

```bash
git config --global user.email "seu.email@exemplo.com"
```

### 💡 O que é o `--global`?

O parâmetro `--global` diz ao Git: "use esse nome e esse e-mail para **todos** os projetos que eu fizer no meu computador."

  * Se você quisesse configurar o nome **apenas para o projeto atual**, o comando seria `git config user.name "SEU NOME"`, sem o `--global`.

### Como Verificar se Funcionou

Depois de rodar os dois comandos acima, você pode verificar se a configuração foi salva corretamente rodando:

```bash
git config --list
```

Seu nome e e-mail devem aparecer na lista.

-----

**Resumindo:** O problema não foi um erro de instalação, mas sim que você parou o comando na metade.

`git config` precisa de uma **ação** (como `user.name` ou `user.email`) para saber o que fazer.

Fica tranquila, errar o comando é o jeito mais rápido de aprender\! Se isso resolveu, me avisa e **me marca como a melhor resposta**\!

**Próximo Passo:** Se precisar de ajuda com os próximos comandos (como `git init` ou `git add`), é só me chamar\!
